### DISCUSSION TITLE
Changing networks

### DISCUSSION THREAD

**Discussion by AranAragan** [TenKeyLabs/dappwright#52] [Q&A] [answered]
Hello!
Does dappwright have the ability to approve when changing networks?

<img width="1272" alt="image" src="https://user-images.githubusercontent.com/62331710/213175265-0f47d6d1-8341-4754-9ec1-eac3899e9a6c.png">

<img width="1276" alt="image" src="https://user-images.githubusercontent.com/62331710/213175640-eb52b57a-0586-4ecb-9f68-38572739baf2.png">

**Comment by osis** [comment, marked as answer]
Yes! Although the that functionality slipped into #29 and I just realized now that it didn't make it into the notes. Thanks for the reminder and we'll make sure a new release goes out today so you can start using it.

**Reply by AranAragan** [reply]
Thank you very much
I'll be looking forward to it)

**Reply by osis** [reply]
Release just went public. Let us know how it goes 👍🏾 

https://www.npmjs.com/package/@tenkeylabs/dappwright/v/2.1.0

**Reply by AranAragan** [reply]
This worked for the Switch Network window, but can't get around add a network

![image](https://user-images.githubusercontent.com/62331710/213384827-e1fd4150-d3d6-4b2c-8834-5223e03c76ee.png)

**Reply by osis** [reply]
Hmmmm that interaction might not be captured at the moment. The flow that the test cases follow is using the `addNetwork()` action and then initiating this flow. That should work for now.

**Reply by AranAragan** [reply]
Unfortunately addNetwork() and approve() do not initialize

**Reply by osis** [reply]
Can you provide a code example to compare it to how your approach differs from the [test scenario](https://github.com/TenKeyLabs/dappwright/blob/cfa10b84c0156af974922ae728f6ca9ebfadf268/test/2-wallet.spec.ts#L77-L99)

**Reply by AranAragan** [reply]
Thank you very much for your help. It turned out to need to add the network at the beginning, before initializing the switch.
